Engine/plugin/agsplugin.cpp

```diff
@@ -977,11 +977,26 @@ void pl_read_plugins_from_disk (Stream *in) {
 
         // load the actual plugin from disk
         EnginePlugin *apl = &plugins[a];
-        strncpy(apl->filename, buffer, strlen(buffer) - strlen(".dll"));
+        
+        // remove extension (examples of extension could be .dll, .dylib, .so, .so.1, etc)
+        {
+            char *ext = strchr(buffer, '.');
+            if (ext) {
+                *ext = 0;
+            }
+        }
+        
+        if (strlen(buffer) > PLUGIN_FILENAME_MAX) {
+            sprintf(buffer, "Plugin '%s' is not a valid AGS plugin because the filename is invalid.", buffer);
+            quit(buffer);
```

Comment:
Yeah, with that sprintf MS runtime produces following message:
<pre>Plugin 'Plugin 'Plugin 'Plugin 'Plugin 'Plugin 'Plugin 'Plugin 'Plugin ' is not a valid AGS plugin because the filename is invalid.</pre>